include/rocksdb/file_checksum.h

```diff
@@ -22,9 +22,13 @@ namespace ROCKSDB_NAMESPACE {
 constexpr char kUnknownFileChecksum[] = "";
 // The unknown sst file checksum function name.
 constexpr char kUnknownFileChecksumFuncName[] = "Unknown";
+// The default DB file checksum function name.
+constexpr char kDefaultDbFileChecksumFuncName[] = "FileChecksumCrc32c";
 
 struct FileChecksumGenContext {
   std::string file_name;
+  // The name of the requested checksum generator
```

Comment:
So that a single factory can create generators for a suite of different functions, like for migrating from one function to another? That seems like interesting new functionality worth mentioning here and in HISTORY.md.

Should a factory have a default function that it returns when this string is empty?

Reply:
Either here or at CreateFileChecksumGenerator, we have to say something about expectations around
(a) backward compatibility, including
(b) if requested_checksum_func_name is empty
(c) if requested_checksum_func_name is not a recognized function that can be provided. Should requested_checksum_func_name return null unique_ptr?

cc @zhichao-cao

Reply:
Still TODO

Reply:
> So that a single factory can create generators for a suite of different functions, like for migrating from one function to another? That seems like interesting new functionality worth mentioning here and in HISTORY.md.
> 
> Should a factory have a default function that it returns when this string is empty?

I think Zippy is trying the similar things. For the same Factory, they use SHA1 as default, but also support crc32c. But I'm not sure how they achieve the switch.

Reply:
> Either here or at CreateFileChecksumGenerator, we have to say something about expectations around
> (a) backward compatibility, including
> (b) if requested_checksum_func_name is empty
> (c) if requested_checksum_func_name is not a recognized function that can be provided. Should requested_checksum_func_name return null unique_ptr?
> 
> cc @zhichao-cao

My question here is, should we set a concrete contract with the user that provide the implementation of ChecksumGenFactory. If so, then, we need to set clear rules to regulate the behavior of CreateFileChecksumGenerator 1) if requested_checksum_func_name is empty, 2) if requested_checksum_func_name does not match. Otherwise, user can define the behavior of their own Factory.